gencred/cmd/main.go

```diff
@@ -0,0 +1,200 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	flag "github.com/spf13/pflag"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/clientcmd/api/latest"
+	gencred "k8s.io/test-infra/gencred/pkg"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	// defaultContextName is the default context name.
+	defaultContextName = "build"
+	// defaultConfigFileName is the default kubeconfig filename.
+	defaultConfigFileName = "/dev/stdout"
+)
+
+// options are the available command-line flags.
+type options struct {
+	context        string
+	name           string
+	output         string
+	certificate    bool
+	serviceaccount bool
+	overwrite      bool
+}
+
+// parseFlags parses the command-line flags.
+func (o *options) parseFlags() {
+	// --context
+	flag.StringVar(&o.context, "context", "", "The name of the kubeconfig context to use.")
+
+	// -n, --name
+	flag.StringVarP(&o.name, "name", "n", defaultContextName, "Context name for the kubeconfig entry.")
+
+	// -o, --output
+	flag.StringVarP(&o.output, "output", "o", defaultConfigFileName, "Output path for generated kubeconfig file.")
+
+	// -c, --certificate
+	flag.BoolVarP(&o.certificate, "certificate", "c", false, "Authorize with a client certificate and key.")
+
+	// -s, --serviceaccount
+	flag.BoolVarP(&o.serviceaccount, "serviceaccount", "s", false, "Authorize with a service account.")
+
+	// --overwrite
+	flag.BoolVar(&o.overwrite, "overwrite", false, "Overwrite (rather than merge) output file if exists.")
+
+	flag.Parse()
+}
+
+// validateFlags validates the command-line flags.
+func (o *options) validateFlags() {
+	var err error
+
+	if len(o.name) == 0 {
+		printErrAndExit("-n, --name option is required.", 1)
+	}
+
+	o.output, err = filepath.Abs(o.output)
+	if err != nil {
+		printErrAndExit(fmt.Sprintf("-o, --output option invalid: %v.", o.output), 1)
```

Comment:
Nit: It is uncommon and unexpected to exit in code that lives outside of the `main` func, returning an error and handling it in main is nicer and allows to write tests

Reply:
Updated. Returning a custom error and only performing the print exit logic in main function.